holoviews/plotting/plotly/raster.py

```diff
@@ -39,6 +42,13 @@ def get_data(self, element, ranges, style):
         if self.invert_axes:
             x0, y0, dx, dy = y0, x0, dy, dx
             array = array.T
+
+        plot_opts = element.opts.get('plot', 'plotly')
+        nodata = plot_opts.kwargs.get('nodata')
+        if nodata is not None and (array.dtype.kind  == 'i'):
+            array = array.astype(np.float64)
+            array[array == nodata] = np.NaN
```

Comment:
This chunk of code occurs so many times in this PR that it really seems to call out for centralizing it somewhere:


```
def apply_nodata(opts, data):
    "Replace `nodata` value in data with NaN, if specified in opts"
    nodata = opts.kwargs.get('nodata')
    if nodata is not None and (data.dtype.kind  == 'i'):
        data = data.astype(np.float64)
        data[data == nodata] = np.NaN
        return data

array = apply_nodata(element.opts.get('plot', 'plotly'), array)
```

Reply:
I considered this but I don't know how helpful it really is: there isn't that much duplication *per backend*. Each time this occurs, you go from 5 lines (where it is very easy to see what it does) to one line where you would need to look up what something like `apply_nodata(element.opts.get('plot', 'plotly'), array)` means, Note there is also a `return data` missing at the end of the suggested function...

Not strongly opposed so I am happy to factor out a utility if @philippjfr thinks it makes sense...

Reply:
Happy to let Philipp be the tiebreaker here as well, but I disagree that it's _very_ easy to see what it does.  At first glance it's a chunk of code that my eyes skip over, not a semantically meaningful action whose intent is obvious when skimming. `apply_nodata` is clearer to me in its intent, even in skimming.  If you study the code line by line, yes, it's easy enough to follow, but that does take work! And then if one wanted to update this code (e.g. if we decided it should apply to floats, or to change the float type), it's very likely that would happen only on one backend or element type, in part because the code isn't identical in the many different spots it appears, due to the name of `data` changing everywhere.  

Anyway, smells bad to me as it is, but @philippjfr can make the call.

Reply:
+1 for factoring it out to `plotting/util.py`

Reply:
Addressed in 2c1e1e0